src/clj/game/core/gaining.clj

```diff
@@ -45,47 +46,49 @@
                    (pos? (get-in @state [:runner :run-credit] 0)))
           (swap! state update-in [:runner :run-credit] (sub->0 value))))))
 
-(defn gain [state side & args]
-  (doseq [[cost-type amount] (partition 2 args)]
-    (cond
-      ;; amount is a map, merge-update map
-      (map? amount)
-      (doseq [[subtype amount] amount]
-        (swap! state update-in [side cost-type subtype] (safe-inc-n amount))
-        (swap! state update-in [:stats side :gain cost-type subtype] (fnil + 0) amount))
+(defn gain [state side eid & args]
+  (effect-seq state side eid [[cost-type amount] (partition 2 args)]
```

Comment:
If the goal of `effect-seq` is to act like `doseq`, we gotta change how it's written cuz this isn't correct for the implementation.